source/installation_guide/upgrading/from-2.x-to-3.0.rst

```diff
@@ -0,0 +1,37 @@
+=====================================
+Upgrading Dovecot from 2.x to 3.0/2.4
+=====================================
+
+.. contents::
+   :depth: 1
+   :local:
+
+
+About version numbers
+=====================
+
+With 3.0 / 2.4 release Pro and CE release have different major version numbers.
+In documentation any reference to 3.0 applies to 2.4 as well.
+
+Recommended upgrade path
+========================
+
+The recommended upgrade path for 3.0/2.4 is to perform a migration to the version.
+We do not recommend in-place upgrade due to significant number of removed and changed
+features. Performing migration is the safest way to start using 3.0/2.4 version.
```

Comment:
Why wouldn't we support in-place migration? This is the first I've heard this. I'm not planning on any incompatible file format changes, except possibly some of the config changes but maybe they can also be handled with old-set-parser.

Reply:
I'm not trying to say it's not supported as per se, just that the path of least resistance is migration.

Reply:
But migration is a very high effort, with a possibility of messing it up and losing data, so I don't see how it can be least resistance. Even recommending it that way will likely cause lots of mails to dovecot-list with people asking about it.